File: playground/devfancy/java-springboot-order-perf-optimizer/src/main/java/com/order/perf/dto/DeliveryDetailsDto.java
package com.order.perf.dto;

import lombok.Getter;

// 배송 정보 DTO
@Getter
public class DeliveryDetailsDto {

    private String recipientName;
    private String mobile;
    private String address;
    private String zipCode;
    private String storePassword;
    private String deliveryMemo;
}

File: playground/ruthetum/app/e-commerce/order/src/main/kotlin/team/sipe/spurt/ecommerce/order/dto/DeliveryDetailsDto.kt
package team.sipe.spurt.ecommerce.order.dto

data class DeliveryDetailsDto(
    val recipientName: String,
    val mobile: String,
    val address: String,
    val zipCode: String,
    val storePassword: String,
    val deliveryMemo: String,
) {
    companion object {
        fun of(
            recipientName: String,
            mobile: String,
            address: String,
            zipCode: String,
            storePassword: String,
            deliveryMemo: String,
        ) = team.sipe.spurt.ecommerce.order.dto.DeliveryDetailsDto(
            recipientName = recipientName,
            mobile = mobile,
            address = address,
            zipCode = zipCode,
            storePassword = storePassword,
            deliveryMemo = deliveryMemo,
        )
    }
}